samples/openthread/ncp/src/user_vendor_hook.cpp

```diff
@@ -0,0 +1,144 @@
+
+namespace ot {
+namespace Ncp {
+
+otError NcpBase::VendorCommandHandler(uint8_t aHeader, unsigned int aCommand)
+{
+    otError error = OT_ERROR_NONE;
+
+    LOG_DBG("VendorCommandHandler");
+
+    switch (aCommand)
+    {
+        // TODO: Implement your command handlers here.
+
+    default:
+        error = PrepareLastStatusResponse(aHeader, SPINEL_STATUS_INVALID_COMMAND);
+    }
+
+    return error;
+}
+
+void NcpBase::VendorHandleFrameRemovedFromNcpBuffer(Spinel::Buffer::FrameTag aFrameTag)
+{
+    // This method is a callback which mirrors `NcpBase::HandleFrameRemovedFromNcpBuffer()`.
+    // It is called when a spinel frame is sent and removed from NCP buffer.
+    //
+    // (a) This can be used to track and verify that a vendor spinel frame response is
+    //     delivered to the host (tracking the frame using its tag).
+    //
+    // (b) It indicates that NCP buffer space is now available (since a spinel frame is
+    //     removed). This can be used to implement reliability mechanisms to re-send
+    //     a failed spinel command response (or an async spinel frame) transmission
+    //     (failed earlier due to NCP buffer being full).
+
+    OT_UNUSED_VARIABLE(aFrameTag);
+}
+
+otError NcpBase::VendorGetPropertyHandler(spinel_prop_key_t aPropKey)
+{
+    otError error = OT_ERROR_NONE;
+
+    switch (aPropKey)
+    {
+    case VENDOR_SPINEL_PROP_VENDOR_NAME:
+        LOG_DBG("Got NAME get property request");
+        error = mEncoder.WriteUtf8("Nordic Seminconductor");
+        break;
+    case VENDOR_SPINEL_PROP_AUTO_ACK_ENABLED:
+        LOG_DBG("Got VENDOR_SPINEL_PROP_AUTO_ACK_ENABLED get property request");
+        error = mEncoder.WriteUint8(nrf_802154_radio_wrapper_auto_ack_get());
+        break;
+    case VENDOR_SPINEL_PROP_HW_CAPABILITIES:
+        LOG_DBG("Got VENDOR_SPINEL_PROP_HW_CAPABILITIES get property request");
+        error = mEncoder.WriteUint16(nrf_802154_radio_wrapper_hw_capabilities_get());
+        break;
+
+        // TODO: Implement your get properties handlers here.
+    default:
+        error = OT_ERROR_NOT_FOUND;
+        break;
+    }
+
+    return error;
+}
+
+otError NcpBase::VendorSetPropertyHandler(spinel_prop_key_t aPropKey)
+{
+    otError error = OT_ERROR_NONE;
+
+    switch (aPropKey)
+    {
+    case VENDOR_SPINEL_PROP_AUTO_ACK_ENABLED:{
+            LOG_DBG("Got VENDOR_SPINEL_PROP_AUTO_ACK_ENABLED set property request");
+
+            uint8_t mode = 0;
+            SuccessOrExit(error = mDecoder.ReadUint8(mode));
+            nrf_802154_radio_wrapper_auto_ack_set(mode);
+        }
+        break;
+
+        // TODO: Implement your set properties handlers here.
+    default:
+        error = OT_ERROR_NOT_FOUND;
+        break;
+    }
+
+exit:
+    return error;
+}
+
+} // namespace Ncp
+} // namespace ot
+
+//-------------------------------------------------------------------------------------------------------------------
```

Comment:
Do we really need such lenghty separator?

Reply:
Done